lib/src/Pos/Logic/Types.hs

```diff
@@ -0,0 +1,171 @@
+{-# LANGUAGE StandaloneDeriving #-}
+
+module Pos.Logic.Types
+    ( LogicLayer (..)
+    , Logic (..)
+    , KeyVal (..)
+    , GetBlockError (..)
+    , GetBlockHeaderError (..)
+    , GetTipError (..)
+    , dummyLogicLayer
+    ) where
+
+import           Universum
+import           Pos.Communication         (NodeId)
+import           Pos.Core.Configuration    (HasGenesisBlockVersionData, genesisBlockVersionData)
+import           Pos.Core.Block            (Block, BlockHeader)
+import           Pos.Core.Types            (HeaderHash, StakeholderId,
+                                            ProxySKHeavy)
+import           Pos.Core.Txp              (TxId, TxAux)
+import           Pos.Core.Update           (UpId, UpdateVote, UpdateProposal, BlockVersionData)
+import           Pos.Ssc.Message           (MCOpening, MCShares, MCCommitment,
+                                            MCVssCertificate)
+
+-- | The interface to a diffusion layer, i.e. some component which takes care
+-- of getting data in from and pushing data out to a network.
+data Logic m = Logic
```

Comment:
Here's the interface so far (may discover other things are needed as I move forward).

Some things to note:
- See comments surrounding the `KeyVal` type for relayed things.
- Adopted `BlockVersionData` must be here, because it's used to implement type-directed message limits.
- Ultimately, the `m` parameter will *not* be `RealMode`. It must be something weaker like `Production`/`IO`. That's because the diffusion layer must use the logic layer features, and diffusion layer must not have `RealMode`.

Reply:
@georgeee